src/frontend/src/lib/components/transactions/TransactionModal.svelte

```diff
@@ -0,0 +1,125 @@
+<script lang="ts">
```

Comment:
Note: this component is not yet fully compatible with ETH and IC. I will be adjusting it for each network separately. In the end, the goal is to re-use as many parts as possible, and the rest will be passed as slots.